test/remote-connector.test.js

```diff
@@ -87,3 +87,81 @@ describe('RemoteConnector', function() {
     });
   });
 });
+
+describe('Custom Path', function() {
+
+  var test = this;
+
+  before(function(done) {
+
+    test.server = server = loopback();
+    server.set('legacyExplorer', false);
+    server.use('/api', loopback.rest());
+    server.set('host', '127.0.0.1');
+    server.set('port', 3001);
```

Comment:
I see.

IIUC, this test case requires a model that is attached to a datasource, instead of mocking the methods as in the other test cases. I don't have a strong opinion which approach is better (mock methods vs. use an in-memory datasource). If it makes more sense to use an in-memory datasource then I'm ok with that.

The other difference I can see is mounting the REST adapter at `/api` vs. `/` - I don't think this is relevant to these tests, your new test case should pass when REST is mounted at `/` too.

So in my view, this lengthy setup can be shared by both test cases, the only difference is whether the `ServerModel` is attached or not. Unless I am missing something else here?